cmake/RMM_thirdparty.cmake

```diff
@@ -22,28 +22,37 @@ CPMAddPackage(
   DOWNLOAD_ONLY TRUE
 )
 
-set(THRUST_INCLUDE_DIR "${thrust_SOURCE_DIR}" PARENT_SCOPE)
+set(THRUST_INCLUDE_DIR "${thrust_SOURCE_DIR}")
 
 ###################################################################################################
 # - googletest ------------------------------------------------------------------------------------
 
 if (BUILD_TESTS)
-  CPMAddPackage(
-    NAME googletest
+  CPMFindPackage(
+    NAME GTest
     GITHUB_REPOSITORY google/googletest
     GIT_TAG release-1.10.0
     VERSION 1.10.0
     GIT_SHALLOW TRUE
     OPTIONS
       "INSTALL_GTEST OFF"
+    # googletest >= 1.10.0 provides a cmake config file -- use it if it exists
+    FIND_PACKAGE_ARGUMENTS "CONFIG"
```

Comment:
Is there a reason we want to avoid `Module` mode for GTest?

Reply:
Yes, module mode doesn't support gmock. 1.10 comes with it's own config, which does support gmock. (And I'd say it seems generally preferable to use the config that comes with a package over a potentially out of date find module).